streams/src/main/java/org/apache/kafka/streams/kstream/SessionWindows.java

```diff
@@ -140,7 +140,7 @@ public static SessionWindows with(final Duration inactivityGap) {
         final String msgPrefix = prepareMillisCheckFailMsgPrefix(inactivityGap, "inactivityGap");
         final long inactivityGapMs = validateMillisecondDuration(inactivityGap, msgPrefix);
 
-        return new SessionWindows(inactivityGapMs, DEPRECATED_OLD_24_HR_GRACE_PERIOD);
+        return new SessionWindows(inactivityGapMs, Math.max(DEPRECATED_OLD_24_HR_GRACE_PERIOD - inactivityGapMs, 0));
```

Comment:
Is this correct? The old `maintainMs` did use `Math.max(grace, inactivityGap)` ?